app/controllers/PushPullNotificationController.scala

```diff
@@ -45,8 +52,13 @@ class PushPullNotificationController @Inject() (cc: ControllerComponents, pushPu
         case Some(clientId) =>
           pushPullNotificationConnector.getBox(clientId).map {
             case Left(error) if error.statusCode == NOT_FOUND =>
-              NotFound(Json.toJson(JsonClientErrorResponse(NOT_FOUND, "No box found for your client id")))
-            case Left(_)    => InternalServerError(Json.toJson(JsonSystemErrorResponse(INTERNAL_SERVER_ERROR, "Unexpected Error")))
+              val errorMessage = Json.toJson(JsonClientErrorResponse(NOT_FOUND, "No box found for your client id"))
```

Comment:
This one you have done in v1 and that is not part of this ticket I believe.

Reply:
it's one of the failed audit defined in the confluence `https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?spaceKey=CTCTA&title=Auditing%3A+Architecture+and+Structure`

Reply:
Sandeep is right though -- we shouldn't be touching v1 in this ticket. All of the auditing we need to do is in v2.

Reply:
Yes I knew that. I think you only had created this list of event with audit spike. As this is for getBox and that part we are doing in ```transit-movements-push-notifications``` for P5. so if you really want to do that then this should go inside ```transit-movements-push-notifications```

Reply:
That's not so true. We can still do that in `common-transit-convention-traders` in response to the call to get the box.

Reply:
ok, will remove this event from v1 code

Reply:
can not see ```PushPullNotificationGetBoxFailed``` with version 2 so still this is not resolved looks like.

Reply:
we get the `BoxResponse[BoxId]` through `postAssociation()`  call for which we already have failed event as `PushNotificationFailed`

Reply:
@drnaylor-oc are you suggesting to implement this ```PushPullNotificationGetBoxFailed``` for ```NOT_FOUND``` case in ```common-transit-convention-traders``` for ```associate()``` method PPNS service call?

Reply:
I just did a search of the code in my IDE:

* `PushNotificationFailed` is defined, but not used anywhere. According to your spikes and what I pulled together at https://confluence.tools.tax.service.gov.uk/display/CTCTA/Auditing%3A+Architecture+and+Structure, this should be when an actual push notification fails. This will mostly happen from the router, but _can_ happen from here too.
* `PushPullNotificationGetBoxFailed` is for when the association fails, so as @sonisn1988 said.

Reply:
`PushNotificationFailed` is defined maybe update the code  - this is added for when association fails.
Will use `PushPullNotificationGetBoxFailed` instead of `PushNotificationFailed` for when the association fails.

So, `PushNotificationFailed` in CTC traders where we should use this?

Reply:
```PushNotificationFailed``` you are going to use for ```InternalServerError``` and ```PushPullNotificationGetBoxFailed``` you are going to use for ```NotFound``` case for the same PPNS call.

Reply:
As mentioned earlier,  the err here is `PushNotificationError` not PresentationError, so we can't check the status code here. 
And also don't like the idea of this as well - if we do separate cases for each and every error code then we have to do this for the rest of the project too. We should just do one generic failed event.

On the other hand, the original intention to add `PushPullNotificationGetBoxFailed` in the spike is for v1 endpoint  as this is not applicable for us.

So,  for `associate()` failed scenario- either we go with PushPullNotificationGetBoxFailed or PushNotificationFailed or some other proper audit event name which is more generic.

Reply:
You will be receiving error type like ```PushNotificationError.BoxNotFound``` and ```PushNotificationError.UnexpectedError``` and then take the decision.  Probably Dan can suggest if we don't want to implement ```PushPullNotificationGetBoxFailed``` here or can suggest.

Reply:
I'd rather see it implemented as was spiked. We agreed on these audit types, let's try to implement them as suggested in the first place.